src/core/public/legacy_platform/legacy_platform_service.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { InjectedMetadataService } from '../injected_metadata';
+
+interface Deps {
+  injectedMetadata: ReturnType<InjectedMetadataService['start']>;
```

Comment:
I'm really not sure what to name this `type`, but once we get plugins in the minx we will probably want to export a named type they can use to define the dependencies of their start functions... `InjectedMetadataContract`? `InjectedMetadataStartContract`? `StartedInjectedMetadata`...

Reply:
`InjectedMetadataStartContract` works for me. I never really intended the word "contract" to be codified, but if that's the language we're using when we talk about this thing, we might as well just write it that way as well.